lib/iris/fileformats/netcdf/_thread_safe_nc.py

```diff
@@ -340,3 +340,41 @@ def __getstate__(self):
     def __setstate__(self, state):
         for key, value in state.items():
             setattr(self, key, value)
+
+
+class DeferredSaveWrapper:
```

Comment:
This has been created as a way to set up the deferred writing, but right now this class is not really doing any "deferred" action, right? So maybe just `SaveWrapper` or something like `NetCDFDataWriter`?